core/ResponseDefinitions.php

```diff
@@ -176,6 +176,34 @@
  *       iconURL: ?string,
  *       iconEmoji: ?string,
  *   }
+ *
+ * @psalm-type CoreTaskProcessingShape = array{
+ *     name: string,
+ *     description: string,
+ *     type: int,
+ *     mandatory: bool,
+ * }
+ *
+ * @psalm-type CoreTaskProcessingTaskType = array{
+ *     name: string,
+ *     description: string,
+ *     inputShape: CoreTaskProcessingShape[],
+ *     outputShape: CoreTaskProcessingShape[],
+ * }
+ *
+ * @psalm-type CoreTaskProcessingTask = array{
+ *     id: int,
+ *     type: string,
+ *     status: 'STATUS_CANCELLED'|'STATUS_FAILED'|'STATUS_SUCCESSFUL'|'STATUS_RUNNING'|'STATUS_SCHEDULED'|'STATUS_UNKNOWN',
+ *     userId: ?string,
+ *     appId: string,
+ *     input: array<string, numeric|list<numeric>|string|list<string>>,
+ *     output: ?array<string, numeric|list<numeric>|string|list<string>>,
```

Comment:
```suggestion
 *     output: null|array<string, numeric|list<numeric>|string|list<string>>,
```
The question mark can be overlooked quickly and for me personally it's hard to parse if it means the first type can also be null or the whole type can be null, so this makes it clear what you want (though both should work).